File: app/src/main/java/com/a6v/tjreader/fragments/BaseFragment.java
package com.a6v.tjreader.fragments;

import android.support.v4.app.Fragment;

import com.a6v.tjreader.activities.AbstractActivityComponent;
import com.a6v.tjreader.activities.ActivityComponentProvider;

public class BaseFragment extends Fragment {
  protected <T extends AbstractActivityComponent> T getActivityComponent() {
    return ((ActivityComponentProvider<T>) getActivity()).getActivityComponent();
  }
}

File: app/src/main/kotlin/com/a6v/tjreader/fragments/BaseFragment.kt
package com.a6v.tjreader.fragments

import android.content.Context
import com.trello.rxlifecycle.components.support.RxFragment

abstract class BaseFragment : RxFragment() {
  override fun onAttach(context: Context) {
    super.onAttach(context)
    injectSelf()
  }

  abstract fun injectSelf()
}